paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt

```diff
@@ -165,6 +177,22 @@ class Paparazzi @JvmOverloads constructor(
 
   fun <V : View> inflate(@LayoutRes layoutId: Int): V = layoutInflater.inflate(layoutId, null) as V
 
+  @JvmOverloads
+  fun snapshot(
+    composable: @Composable () -> Unit,
+    name: String? = null,
+    deviceConfig: DeviceConfig? = null,
+    theme: String? = null,
+    renderingMode: RenderingMode? = null
```

Comment:
Can I suggest moving `composable` to the last param? This would let us write tests in this manner:

```kotlin
paparazzi.snapshot(name = "foo") {
  Text("whoa")
}
```

Reply:
I was opting for consistency, but you raise a fair point.  Is it a weird api to have View be the first arg in legacy Android snapshots, while a Composable be the last arg?

Reply:
I actually think they are both consistent because I can call `paparazzi.snapshot` without any extra configuration:

```kotlin
paparazzi.snapshot(ImageView())

paparazzi.snapshot {
  Image()
}
```

Reply:
What about the cases where extra configuration is preferred?

As is, without configuration:
```
paparazzi.snapshot(ImageView())
paparazzi.snapshot({Image()})
```

As is, with configuration:
```
paparazzi.snapshot(ImageView(), name = "foo")
paparazzi.snapshot({Image()}, name = "foo")
```

Proposed, without configuration:
```
paparazzi.snapshot(ImageView())
paparazzi.snapshot { Image() }
```

Proposed, with configuration:
```
paparazzi.snapshot(ImageView(), name = "foo")
paparazzi.snapshot(name = "foo") { Image() }
```

Of course, we can move View to the last param, but that also will break everyone's current tests...

Reply:
![image](https://user-images.githubusercontent.com/2387680/154162857-92294b48-af7f-4f24-b03f-9e7215c2cb05.png)

I'd say this is good enough! Both these APIs are in line with their respective worlds.

Reply:
Or it could be fluent:

```
interface PaparazziConfiguration {
  fun deviceConfig(deviceConfig: DeviceConfig): PaparazziConfiguration
  fun theme(theme: String): PaparazziConfiguration
  fun renderingMode(renderingMode: RenderingMode): PaparazziConfiguration

  fun snapshot(view: View)
  fun snapshot(composable: @Composable ()->Unit)
}
```

Has the advantage of providing a smaller ABI surface area on each method call!